test/src/main/java/org/apache/accumulo/test/ComprehensiveIT.java

```diff
@@ -133,6 +153,51 @@ public static void teardown() {
     SharedMiniClusterBase.stopMiniCluster();
   }
 
+  @Test
+  public void testEventualScan() throws Exception {
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      String tableName = getUniqueNames(1)[0];
+      client.tableOperations().create(tableName);
+
```

Comment:
This test has some utils that are use to read and write data.  Using those will test authorizations, scanner, and batch scanner.  Also would make the code shorter in addition to testing more.  The test functions allow passing a lambda to configure the scanner can use that to set eventual consistency.

Could use them in this test like the following to create a basic verification of eventual scans.

```java
      // TODO start scan server

      // generate 300 rows of data
      write(client, table, generateMutations(0, 300, tr -> true));
      
      // eventual scan should not see anything when table is not flushed 
      verifyData(client, table, AUTHORIZATIONS, Map.of(),
          scanner -> scanner.setConsistencyLevel(EVENTUAL));

      // TODO need flush table so that eventual scan can see it.

      // should see all data that was flushed in eventual scan
      verifyData(client, table, AUTHORIZATIONS, generateKeys(0, 300),
          scanner -> scanner.setConsistencyLevel(EVENTUAL));
      // should not see data with col vis set
      verifyData(client, table, Authorizations.EMPTY, generateKeys(0, 300, tr -> tr.vis.isEmpty()),
          scanner -> scanner.setConsistencyLevel(EVENTUAL));

     // TODO write some more rows after 300 and verfy those are not seen by eventual scan until table is flushed.
```

Reply:
My suggestion has a bug. If you run eventual scan before flush, then run eventual scan after flush will probably still see no data as the scan server caches tablet files for a bit.   So would either need to drop that scan before flush, or set the scan server cache timeout low and wait for longer than that timeout before scanning after flush.